integration-tests/src/test/java/com/datastax/oss/driver/api/core/cloud/CloudIT.java

```diff
@@ -219,4 +222,23 @@ public void should_error_when_contact_points_and_secure_bundle_used() {
         .hasMessage(
             "Can't use withCloudSecureConnectBundle and addContactPoint(s). They are mutually exclusive.");
   }
+
+  @Test
+  public void should_error_when_ssl_context_and_secure_bundle_used() {
+    // given
+    try {
+      Path bundle = proxyRule.getProxy().getBundleWithoutCredentialsPath();
+      CqlSessionBuilder builder =
+          CqlSession.builder()
+              .withCloudSecureConnectBundle(bundle)
+              .withAuthCredentials("cassandra", "cassandra")
+              .withSslContext(SSLContext.getInstance("SSL"));
+      assertThatThrownBy(() -> builder.build())
+          .isInstanceOf(IllegalStateException.class)
+          .hasMessage(
+              "Can't use withCloudSecureConnectBundle and addContactPoint(s). They are mutually exclusive.");
+    } catch (NoSuchAlgorithmException e) {
```

Comment:
Maybe we could pick an existing `SSLContext` https://github.com/datastax/java-driver/pull/1373/files#diff-d001c2e83a104c3b75049085a3425839R235 to not have that `NoSuchAlgorithmException`? It makes this test a little bit less readable.

Reply:
I think the mistake was to catch `NoSuchAlgorithmException`: it would be better to simply have the test method declare to throw it imo.